contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,279 @@
  * SPDX-License-Identifier: UNLICENSED
  */
 pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import {Initializable} from "./packages/oz/upgradeability/Initializable.sol";
+import {SafeMath} from "./packages/oz/SafeMath.sol";
+import {OtokenInterface} from "./interfaces/OtokenInterface.sol";
+import {OracleInterface} from "./interfaces/OracleInterface.sol";
+import {AddressBookInterface} from "./interfaces/AddressBookInterface.sol";
+import {FixedPointInt256} from "./libs/FixedPointInt256.sol";
+import {MarginAccount} from "./libs/MarginAccount.sol";
 
 /**
  *
  */
-// solhint-disable-next-line no-empty-blocks
-contract MarginCalculator {
+contract MarginCalculator is Initializable {
+    using SafeMath for uint256;
+    using FixedPointInt256 for int256;
+    address public addressBook;
+
+    function init(address _addressBook) external initializer {
+        addressBook = _addressBook;
+    }
+
+    /**
+     * @notice Return the cash value of an expired oToken.
+     * @dev For call return = Max (0, ETH Price - oToken.strike)
+     * @dev For put return Max(0, oToken.strike - ETH Price)
+     * @param _otoken otoken address
+     * @return the cash value of an expired otoken
+     */
+    function getExpiredCashValue(address _otoken) public view returns (uint256) {
```

Comment:
We should leave it as getExpiredCashValue, because if not we need to have the oracle price before expiry at each timestamp right? Is our oracle contract designed to do that? If not, better to leave it as is?

Reply:
Yeah you're right, we don't have the oracle price until expiry.